scalding-core/src/main/scala/com/twitter/scalding/TupleConverter.scala

```diff
@@ -51,6 +53,37 @@ trait LowPriorityTupleConverters extends java.io.Serializable {
     }
 }
 
+/**
+ * Based on `FromTraversable` from shapeless
+ */
+trait FromTupleEntry[Out <: HList, I <: Nat] {
+  def apply(l : TupleEntry) : Option[Out]
+}
+
+object FromTupleEntry {
+  def apply[Out <: HList](implicit from: FromTupleEntry[Out, Nat._0]) = from
+
+  implicit def hnilFromTupleEntry[T, I <: Nat](implicit toIntI : ToInt[I]) =
+    new FromTupleEntry[HNil, I] {
+      def apply(te : TupleEntry) =
+        if(te.size() == toIntI()) Some(HNil) else None
```

Comment:
This forces the TupleEntry to have exactly the same size as the HList.  I'm not sure if it needs to be this strict or if it would be ok for the TupleEntry to have more entries in it than the HList.